src/formatting.js

```diff
@@ -0,0 +1,363 @@
+        var formatNumber;
+        var formatFormat;
+        var result;
+        var base;
+        var value;
+        var handleWidth;
+        var alternateForm;
+        var precedeWithSign;
+        var blankBeforePositive;
+        var leftAdjust;
+        var centerAdjust;
+        var zeroPad;
+        var convName;
+        var convValue;
+        var percent;
+        fieldWidth = Sk.builtin.asnum$(fieldWidth);
+        precision = Sk.builtin.asnum$(precision);
+
+        if(element_index !== undefined && element_index !== ""){
+            value = arg_dict[arg_name][element_index].v;
+            index++;
+        } else if(attribute_name !== undefined && attribute_name !== ""){
+            value = arg_dict[arg_name][attribute_name].v;
+            index++;
+        }
+        else if(arg_name !== undefined && arg_name !== ""){
+            value = arg_dict[arg_name];
+            index++;
+        }
+
+         else if(field_name === undefined || field_name === ""){
+            return_str = arg_dict[index];
+            index++;
+            value = return_str;
+        }
+        else if(field_name instanceof Sk.builtin.nmber || field_name instanceof Sk.builtin.lng || !isNaN(parseInt(field_name, 10))){
+           return_str = arg_dict[field_name];
+           index++;
+           value = return_str;
+        }
+
+        if (precision === "") { // ff passes '' here aswell causing problems with G,g, etc.
+            precision = undefined;
+        }
+        if(fill_char === undefined || fill_char === ""){
+            fill_char = " ";
+        }
+
+        zeroPad = false;
+        leftAdjust = false;
+        centerAdjust = false;
+        blankBeforePositive = false;
+        precedeWithSign = false;
+        alternateForm = false;
+        if (format_spec) {
+            if(sign !== undefined && sign !== ""){
+                if ("-".indexOf(sign) !== -1) {
+                    leftAdjust = true;
+                }
+                else if ("+".indexOf(sign) !== -1) {
+                    precedeWithSign = true;
+                }
+                else if (" ".indexOf(sign) !== -1) {
+                    blankBeforePositive = true;
+                }
+            }
+            if(zero_pad){
+            alternateForm = "#".indexOf(zero_pad) !== -1;
+            }
+            if(fieldWidth !== undefined && fieldWidth !== ""){
+                if(fill_char === undefined || fill_char === ""){
+                    fill_char = " ";
+                }
+            }
+            if("%".indexOf(conversionType) !== -1){
+                percent = true;
+            }
+        }
+        if (precision) {
+            precision = parseInt(precision, 10);
+        }
+
+        formatFormat = function(value){
+            var r;
+            if(conversion === undefined || conversion === ""){
+                return value;
+            }
+            else if( conversion == "r"){
+                r = Sk.builtin.repr(value);
```

Comment:
The problem may be that by the time you are here, value is a raw javascript string.  It is no longer a python object.  ``repr`` does not deal with javascript objects, hence it returns unknown.